File: tcp-socket/node-server/simple_server.js
const http = require('http');

// HTML-контент для ответа
const htmlContent = `
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <title>Simple Web Page</title>
</head>
<body>
    <h1>Hello from Node.js HTTP Server!</h1>
    <p>This is a simple response from a Node.js server.</p>
</body>
</html>
`;

// Настройки сервера
const host = '0.0.0.0';
const port = 8080;

// Создаем сервер
const server = http.createServer((req, res) => {
    // Устанавливаем заголовок и статус ответа
    res.writeHead(200, { 'Content-Type': 'text/html' });
    // Отправляем HTML-контент
    res.end(htmlContent);
});

// Запуск сервера
server.listen(port, host, () => {
    console.log(`Server started at http://${host}:${port}`);
});

File: tcp-socket/python-server/simple_server.py
from http.server import SimpleHTTPRequestHandler, HTTPServer

class SimpleHandler(SimpleHTTPRequestHandler):
    def do_GET(self):
        # Устанавливаем код ответа 200 (ОК)
        self.send_response(200)
        # Указываем, что будет возвращен HTML-контент
        self.send_header("Content-type", "text/html")
        self.end_headers()
        
        # HTML-контент страницы
        html_content = b"""
        <!DOCTYPE html>
        <html lang="en">
        <head>
            <meta charset="UTF-8">
            <title>Simple Web Page</title>
        </head>
        <body>
            <h1>Hello from Python HTTP Server!</h1>
            <p>This is a simple response from a Python server.</p>
        </body>
        </html>
        """
        
        # Отправляем HTML-контент в качестве ответа
        self.wfile.write(html_content)

# Настройки сервера
host = "localhost"
port = 8080

# Создаем и запускаем сервер
with HTTPServer((host, port), SimpleHandler) as server:
    print(f"Server started at http://{host}:{port}")
    server.serve_forever()
